nilearn/decoding/tests/test_decoder.py

```diff
@@ -0,0 +1,286 @@
+"""
+Test the decoder module
+"""
+
+# Author: Andres Hoyos-Idrobo
+#         Binh Nguyen
+#         Thomas Bazeiile
+#
+# License: simplified BSD
+
+import warnings
+
+import sklearn
+import numpy as np
+from nilearn.decoding.decoder import (_BaseDecoder, Decoder, DecoderRegressor,
+                                      _check_estimator, _check_param_grid, 
+                                      _parallel_fit)
+from nilearn.decoding.tests.test_same_api import to_niimgs
+from nilearn.input_data import NiftiMasker
+from nose.tools import assert_equal, assert_raises, assert_true, assert_warns
```

Comment:
We are moving to pytest. @kchawla-pi : should we address the move in this PR, or after merging?

Reply:
I am addressing this in a separate PR.
If this one gets merged first, I can handle the switch in my PR.
If that one gets merged first we'll handle it here.

@tbng can focus on this PR and leave the maintenance stuff like nose to pytest to me.